apps/hash-graph/lib/graph/src/store/postgres/knowledge/entity/read.rs

```diff
@@ -64,16 +64,28 @@ impl<C: AsClient> crud::Read<Entity> for PostgresStore<C> {
 
         let mut compiler = SelectCompiler::new(temporal_axes);
 
-        let owned_by_id_index = compiler.add_selection_path(&EntityQueryPath::OwnedById);
-        let entity_uuid_index = compiler.add_selection_path(&EntityQueryPath::Uuid);
-        let edition_id_index = compiler.add_distinct_selection_with_ordering(
-            &EntityQueryPath::EditionId,
+        let owned_by_id_index = compiler.add_distinct_selection_with_ordering(
+            &EntityQueryPath::OwnedById,
+            Distinctness::Distinct,
+            None,
+        );
+        let entity_uuid_index = compiler.add_distinct_selection_with_ordering(
+            &EntityQueryPath::Uuid,
+            Distinctness::Distinct,
+            None,
+        );
+        let decision_time_index = compiler.add_distinct_selection_with_ordering(
+            &EntityQueryPath::DecisionTime,
+            Distinctness::Distinct,
+            None,
+        );
+        let transaction_time_index = compiler.add_distinct_selection_with_ordering(
```

Comment:
Do we know if the `entity_temporal_metadata_overlapping` GIST index would help alleviate some of the overhead of multi-column `DISTINCT` queries? I assume it would help here, but I can imagine the query plan for this being slightly different now. We could measure/look at the `EXPLAIN` for the two queries to compare

Reply:
I was about to investigate this at some point but didn't do this as part of this PR as the previous implementation was wrong.

Reply:
Yeah, good point. It doesn't matter if the performance is worse if the query was wrong before :smile: